src/main/java/ch/naviqore/service/impl/ConnectionImpl.java

```diff
@@ -16,4 +16,43 @@ public class ConnectionImpl implements Connection {
 
     private final List<Leg> legs;
 
+    @Override
+    public LocalDateTime getDepartureTime() {
```

Comment:
Still not convinced that the visitor is the better solution, than just having the implementation of PublicTransit back referencing the stop time. But I know I'm the minority disagreeing with design patterns, so I can live with it.

Reply:
What bothers you about the visitor? we have no casts and no null values... so it is completely safe ;)

But I know, you really enjoy passing around null values, haha sorry 😄